proto/spire/api/server/local_authority/v1/local_authority.proto

```diff
@@ -0,0 +1,174 @@
+        // which will cause it to be used for all X.509 signing operations
+        // serviced by this server going forward. If a new X.509 authority has
+        // not already been prepared, a FailedPrecondition error will be returned.
+        rpc ActivateX509Authority(ActivateX509AuthorityRequest) returns (ActivateX509AuthorityResponse);
+
+        // TaintX509Authority marks the previously active X.509 authority as
+        // being tainted. SPIRE Agents observing an authority to be tainted
+        // will perform proactive rotations of any key material related to
+        // the tainted authority. The result of this action will be observed
+        // cluster-wide.
+        //
+        // If a previously active X.509 authority does not exist (e.g. if one
+        // has been prepared but not activated yet), a FailedPrecondition
+        // error will be returned.
+        rpc TaintX509Authority(TaintX509AuthorityRequest) returns (TaintX509AuthorityResponse);
+
+        // RevokeX509Authority revokes the previously active X.509 authority by
+        // removing it from the bundle and propagating this update throughout
+        // the cluster.
+        //
+        // If a previously active X.509 authority does not exist (e.g. if one
+        // has been prepared but not activated yet), a FailedPrecondition
+        // error will be returned.
+        rpc RevokeX509Authority(RevokeX509AuthorityRequest) returns (RevokeX509AuthorityResponse);
+}
+
+message GetJWTAuthorityStateRequest {}
+
+message GetJWTAuthorityStateResponse {
+        repeated AuthorityState states = 1;
+}
+
+message PrepareJWTAuthorityRequest {}
+
+message PrepareJWTAuthorityResponse {
+        AuthorityState prepared_authority = 1;
+}
+
+message ActivateJWTAuthorityRequest {}
+
+message ActivateJWTAuthorityResponse {
+        AuthorityState activated_authority = 1;
+}
+
+message TaintJWTAuthorityRequest {}
+
+message TaintJWTAuthorityResponse {
+        AuthorityState tainted_authority = 1;
+}
+
+message RevokeJWTAuthorityRequest {}
+
+message RevokeJWTAuthorityResponse {
+        AuthorityState revoked_authority = 1;
+}
+
+message GetX509AuthorityStateRequest {}
+
+message GetX509AuthorityStateResponse {
+        repeated AuthorityState states = 1;
+}
+
+message PrepareX509AuthorityRequest {}
+
+message PrepareX509AuthorityResponse {
+        AuthorityState prepared_authority = 1;
+}
+
+message ActivateX509AuthorityRequest {}
+
+message ActivateX509AuthorityResponse {
+        AuthorityState activated_authority = 1;
+}
+
+message TaintX509AuthorityRequest {}
+
+message TaintX509AuthorityResponse {
+        AuthorityState tainted_authority = 1;
+}
+
+message RevokeX509AuthorityRequest {}
+
+message RevokeX509AuthorityResponse {
+        AuthorityState revoked_authority = 1;
+}
+
+message AuthorityState {
+        // An opaque name for the "slot" that this authority is housed in
+        string slot = 1;
```

Comment:
I'm not sure we'll be able to fill this for anything but empty, prepared, and active, particularly if we remove the journal as we've been discussing. My gut feeling says that the slot is an internal detail and probably shouldn't be exposed.

Reply:
I'm the same way, right now we has A or B values, that we use as slotID, based in status, we can avoid this slotID and relly o state, so we can have only one PREPARED | ACTIVE  | OLD, 
I'm not sure about the use of EXPIRED, what is the difference? may we just remove expired authorities?